src/helpers/LabelFilterHelper.ts

```diff
@@ -4,36 +4,91 @@ import { ServiceListItem } from '../types/ServiceList';
 
 type itemsType = AppListItem | ServiceListItem | WorkloadListItem;
 
-export const filterByLabel = (items: itemsType[], filter: string[]): itemsType[] => {
-  let result: itemsType[] = [];
-  filter.map(filter => {
-    if (filter.includes(':')) {
-      const values = filter.split(':');
-      // Check Values
-      values[1]
-        .split(',')
-        .map(
-          val =>
-            (result = result.concat(
-              items.filter(
-                item =>
-                  values[0] in item.labels &&
-                  item.labels[values[0].trim()]
-                    .split(',')
-                    .some(appVal => appVal.trim().startsWith(val.trim()) && !result.includes(item))
-              )
-            ))
-        );
-    } else {
-      // Check if has Label
-      result = result.concat(
-        items.filter(item =>
-          Object.keys(item.labels).some(key => key.startsWith(filter.trim()) && !result.includes(item))
-        )
-      );
+const filterLabelByOp = (labels: { [key: string]: string }, filters: string[], op: string = 'or'): boolean => {
+  let filterOkForLabel: boolean = false;
+  // keys => List of filters with only Label Presence
+  // keyValues => List of filters with Label and value
+  /*
+    TS Error but this works...
+  */
+  // const [keys, keyValues] = filters.reduce(([p, f], e) => (!e.includes(':') ? [[...p, e], f] : [p, [...f, e]]), [[], []]);
+  const keys = filters.filter(f => !f.includes(':'));
+  const keyValues = filters.filter(f => f.includes(':'));
+
+  // Get all keys of labels
+  const labelKeys = Object.keys(labels);
+  // Case OR operation
+  if (op === 'or') {
```

Comment:
for easier reading it would be nice to split this into two different functions: one for the "and" case, one for the "or".